app/models/aws_engine.rb

```diff
@@ -0,0 +1,274 @@
+    yield config
+
+    if self.configured?
+      Aws.config.update({
+        region: @config.region,
+        credentials: Aws::Credentials.new(@config.aws_access_id, @config.aws_secret_key),
+      },)
+    end
+  end
+
+  def self.config
+    @config ||= Config.new
+  end
+
+  def self.configured?
+    return false unless defined? @config.aws_access_id
+    return false unless defined? @config.aws_secret_key
+    return false if @config.aws_access_id.blank?
+    return false if @config.aws_secret_key.blank?
+    true
+  end
+
+  def self.resource_endpoint
+    unless @config.cdn_base_url.blank?
+      url = @config.cdn_base_url
+    else
+      if @config.use_s3_static_hosting == '1'
+        url = "http://#{@config.image_bucket_name}"
+      else
+        if @config.region == 'us-east-1'
+          url = "https://#{@config.image_bucket_name}.s3.amazonaws.com"
+        else
+          url = "https://#{@config.image_bucket_name}.s3-#{@config.region}.amazonaws.com"
+        end
+      end
+    end
+    url
+  end
+
+  def self.upload_endpoint
+    if @config.region == 'us-east-1'
+      url = "https://#{@config.bucket_name}.s3.amazonaws.com"
+    else
+      url = "https://#{@config.bucket_name}.s3-#{@config.region}.amazonaws.com"
+    end
+    url
+  end
+
+  def self.s3_host_name
+    if @config.region == 'us-east-1'
+      s3_host_name = 's3.amazonaws.com'
+    else
+      s3_host_name = "s3-#{@config.region}.amazonaws.com"
+    end
+    s3_host_name
+  end
+
+  ## SQS
+  def self.sqs
+    @sqs ||= Aws::SQS::Client.new(region: @config.region)
+  end
+
+  def self.send_message(message)
+    resp = self.sqs.send_message({
+      queue_url: @config.sqs_url,
+      message_body: message,
+    },)
+    resp
+  end
+
+  def self.receive_message(max_number = 10)
+    resp = self.sqs.receive_message(queue_url: @config.sqs_url, visibility_timeout: 600, max_number_of_messages: max_number)
+    resp
+  end
+
+  def self.message_exist?(resp)
+    if !resp || resp.messages.count == 0
+      false
+    else
+      true
+    end
+  end
+
+  def self.delete_message(message_object)
+    resp = self.sqs.delete_message({
+      queue_url: @config.sqs_url,
+      receipt_handle: message_object.receipt_handle,
+    },)
+    resp
+  end
+
+  def self.batch_delete(entries)
+    self.sqs.delete_message_batch(queue_url: @config.sqs_url, entries: entries)
+  end
+
+  ## S3
+  #
+  def self.upload_files(bucket, files, prefix)
+    files.each do |f|
+      Aws::S3::Client.new(region: @config.region).put_object(
+        bucket: bucket,
+        key: "#{prefix}/#{File.basename(f)}",
+        body: File.read(f),
+        acl: 'public-read',
+        storage_class: 'REDUCED_REDUNDANCY',
+      ) if File.exist?(f)
+    end
+  end
+
+  def self.get_file_list(bucket, prefix)
+    resp = Aws::S3::Client.new(region: @config.region).list_objects({
+      bucket: bucket,
+      max_keys: 1000,
+      prefix: prefix,
+    },)
+    files = []
+    resp.contents.each do |f|
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'f' [:ref](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)